geometry/create-and-edit-geometries/src/main/java/com/esri/samples/create_and_edit_geometries/CreateAndEditGeometriesController.java

```diff
@@ -0,0 +1,591 @@
+    geometryEditor.undo();
+  }
+
+  /**
+   * Redo the last change made to the geometry while editing is active.
+   */
+  @FXML
+  private void handleRedoButtonClicked() {
+    geometryEditor.redo();
+  }
+
+  /**
+   * Deletes the currently selected element of the geometry.
+   */
+  @FXML
+  private void handleDeleteSelectedElementButtonClicked() {
+    geometryEditor.deleteSelectedElement();
+  }
+
+  /**
+   * Saves the geometry in the editor to the graphics overlay and sets its symbol to the type relevant for the geometry.
+   */
+  @FXML
+  private void handleStopAndSaveButtonClicked() {
+    if (geometryEditor.getGeometry().isEmpty()) {
+      System.err.println("Geometry is empty. Unable to save geometry.");
+    }
+
+    // handle saving depending on if the geometry being edited is new, or already exists and needs updating
+    if (selectedGraphic != null) {
+      updateSelectedGraphic();
+    } else {
+      createNewGraphic();
+    }
+
+    // enable tool selection once a graphic is no longer being edited
+    toolComboBox.setDisable(false);
+  }
+
+  /**
+   * Adds a new graphic to the graphics overlay.
+   */
+  private void createNewGraphic() {
+    // get the geometry from the geometry editor and create a new graphic for it
+    var geometry = geometryEditor.stop();
+    var graphic = new Graphic(geometry);
+
+    // set graphic style based on geometry type
+    switch (geometry.getGeometryType()) {
+      case POINT:
+        graphic.setSymbol(pointSymbol);
+        break;
+      case MULTIPOINT:
+        graphic.setSymbol(multipointSymbol);
+        break;
+      case POLYLINE:
+        graphic.setSymbol(lineSymbol);
+        break;
+      case POLYGON:
+        graphic.setSymbol(fillSymbol);
+        break;
+    }
+
+    // add new graphic to the graphics overlay
+    graphicsOverlay.getGraphics().add(graphic);
+  }
+
+  /**
+   * Updates a graphic already in the graphics overlay.
+   */
+  private void updateSelectedGraphic() {
+    // update the geometry of the graphic being edited and make it visible again
+    selectedGraphic.setGeometry(geometryEditor.stop());
+    selectedGraphic.setVisible(true);
+    selectedGraphic = null;
+  }
+
+  /**
+   * Stops the geometry editor without saving the geometry stored within.
+   */
+  @FXML
+  private void handleStopAndDiscardButtonClicked() {
+    // if editing a pre-existing graphic, make it visible again
+    if (selectedGraphic != null) {
+      selectedGraphic.setVisible(true);
+      selectedGraphic = null;
+      toolComboBox.setDisable(false);
+    }
+    geometryEditor.stop();
+  }
+
+  /**
+   * Removes all geometries from the graphics overlay.
+   */
+  @FXML
+  private void handleDeleteAllGeometriesButtonClicked() {
+    graphicsOverlay.getGraphics().clear();
+  }
+
+  /**
+   * Allows the user to select a graphic from the graphics overlay and edit its geometry.
+   */
+  private void selectGraphic() {
+
+    mapView.setOnMouseClicked(e -> {
```

Comment:
```suggestion
    mapView.setOnMouseClicked(e -> {
      if (e.getButton() == MouseButton.PRIMARY && e.isStillSincePress()) {
```
Additional checks to avoid unwanted clicks e.g. currently if you drag over a graphic you select something.